Mathlib/Combinatorics/SimpleGraph/Hasse.lean

```diff
@@ -235,19 +234,24 @@ theorem Walk.toPathGraphHom (G : SimpleGraph α) :
           rw [hab] at h'
           exact ((pathGraph (p.length + 2)).loopless a h').elim
       let hom : pathGraph (p.length + 2) →g G := ⟨toFun, map_rel'⟩
-      apply Exists.intro hom
-      have hhom : ∀ (a : Fin (p.length + 2)), hom a = toFun a := fun a ↦ rfl
-      simp only [length]
-      rw [hhom ⊥, hhom ⊤]
-      simp [toFun]
-      apply And.intro
-      · have hbot : (⊥ : Fin (p.length + 2)).val < p.length + 1 := Nat.compare_eq_gt.mp rfl
-        simp [hbot]
-        simp_rw [← hw]
-        exact congrArg hom' rfl
-      · intro h_
-        have htop : (⊤ : Fin (p.length + 2)).val = p.length + 1 := rfl
-        rw [htop] at h_
-        apply (Nat.lt_irrefl (p.length + 1)).elim h_
+      have hhom : hom ⊥ = w ∧ hom ⊤ = u := by
+        have hhom' : ∀ (a : Fin (p.length + 2)), hom a = toFun a := fun a ↦ rfl
+        simp only [length]
+        rw [hhom' ⊥, hhom' ⊤]
+        simp [toFun]
+        apply And.intro
+        · have hbot : (⊥ : Fin (p.length + 2)).val < p.length + 1 := Nat.compare_eq_gt.mp rfl
+          simp [hbot]
+          simp_rw [← hw]
+          exact congrArg hom' rfl
+        · intro h_
+          have htop : (⊤ : Fin (p.length + 2)).val = p.length + 1 := rfl
+          rw [htop] at h_
+          apply (Nat.lt_irrefl (p.length + 1)).elim h_
+      exact ⟨hom, hhom⟩
+
+def Walk.toPathGraphHom (G : SimpleGraph α) :
+    ∀ {u v : α} (w : G.Walk u v), pathGraph (w.length + 1) →g G :=
+  fun {u v} w => (@Walk.toPathGraphHomAux α G u v w).fst
```

Comment:
```suggestion
def Walk.toPathGraphHom (G : SimpleGraph α) {u v : α} (w : G.Walk u v) :
    pathGraph (w.length + 1) →g G := w.toPathGraphHomAux.1
```
and you should also add `Walk.toPathGraphHom_bot : w.toPathGraphHom ⊥ = v := w.toPathGraphHomAux.2.1` and `Walk.toPathGraphHom_top : w.toPathGraphHom ⊤ = v := w.toPathGraphHomAux.2.2`